DEVELOPER_STARTING_GUIDE.md

```diff
@@ -0,0 +1,185 @@
+11. **Start with the jest tests from Shoelace.** Inside of `packages/components` you can test individual components with `pnpm test.component input` (where input is akin to the sd-input component). These will likely contain some misnamed properties or improper default values. Adapt them as needed to fit our use case.
+12. **Add any tests that are needed for new component behavior.** Motto: “Test the behavior, not the implementation.” Write tests that verify your components meet the expected requirements and specifications, ensuring they function correctly from the user's perspective. Don’t get too tied up trying to test all the technical details.
+13. **Author the full migration guide.** The migration guide should be placed in the `packages/components/src/docs/Migration` folder. The migration guide should be based on the migration guide template (`templates/migration-guide-template.mdx`) and be named by the old component name (e.g. ui-button.mdx).
+
+**Run `pnpm verify` in the root directory before creating a PR.** This will check that all formatting, tests, and build processes are working correctly to allow the pipeline to run successfully.
+
+### Styles
+
+If you are working on a style, follow the steps below:
+...
+
+## Repository Overview
+
+- `packages`
+  - `components`
+    - `src`
+      - `_components`: components previously copied from Shoelace that have replace `sl` prefixes with `sd` prefixes
+      - **`components`**: Primary folder containing our Web Components. Added as stories in the "COMPONENTS" section.
+        - `accordion`: Lit JS web component that implements an accordion. The comments are automatically compiled into the Storybook docs and should be used for user clarification.
+        - `accoridion.tests.ts`: Suite of jest tests for the accordion component. Critical for deployment and maintenance.
+        - `accordion.stories.ts`: Collection of Storybook stories that are primarily used for visual testing with Chromatic. Any **samples** are added as stories here.
+      - `docs`: all ".mdx" doc files that appear in the "DOCS" section of our Storybook
+        - `migration`: Individual component migration guides
+      - `internal`: A set of utilities that are reusable internally. Check here should you need something like a debounce, it may have already been implemented.
+      - **`patterns`**: Demonstrate how to combine several components to solve a specific problem. Added as stories in the "PATTERNS" section.
+      - **`styles`**: These are standalone CSS files. They don't provide any logic and are often non-interactive. Added as stories in the "STYLES" section.
+
+## Technologies
+
+- pnpm
+- Web Components
+- Lit JS
+- Tailwind
+- Storybook
+- Jest
+- Chromatic
+- Vite
+- PostCSS
+
+//TBD: Do we keep them and apply them even thought the nome does not come from the SOLID principles?
```

Comment:
comment accidentally left in?

Reply:
Left here on purpose for discussion. Especially for @yoezlem and @mariohamann...;)

Reply:
I would like to keep it. Both the principles with their foundations and the name fit perfectly with us. This guide is written for developers who are not familiar with our stack, our way of working, and our principles. For me as a new developer, it would be helpful.

Reply:
I would like to keep it. Both the principles with their foundations and the name fit perfectly with us. This guide is written for developers who are not familiar with our stack, our way of working, and our principles. For me as a new developer, it would be helpful.

Reply:
```suggestion
```